src/backend/utils/storage.js

```diff
@@ -25,35 +26,40 @@ module.exports = {
    * Feeds
    */
   addFeed: async (feed) => {
-    const key = createFeedKey(feed.id);
-    await redis
-      .multi()
-      // Using hmset() until hset() fully supports multiple fields:
-      // https://github.com/stipsan/ioredis-mock/issues/345
-      // https://github.com/luin/ioredis/issues/551
-      .hmset(
-        key,
-        'id',
-        feed.id,
-        'author',
-        feed.author,
-        'url',
-        feed.url,
-        'user',
-        feed.user,
-        'link',
-        feed.link,
-        'etag',
-        feed.etag,
-        'lastModified',
-        feed.lastModified
-      )
-      .sadd(feedsKey, feed.id)
-      .exec();
+    // Check if feed being added already exists in flagged feeds set
+    if (!(await redis.sismember(flaggedFeedsKey, feed.id))) {
```

Comment:
Nit: if you have to indent your entire function, flip your logic and return early.  I won't block on this, but get into the habit of keeping your code left-aligned.